### ISSUE TITLE
Question: Coloring on Shadowlands Skin

### ISSUE THREAD

**Issue by nancikennedy** [SFX-WoW/Masque#269]
### Game Flavor

Retail

### Game Version

10.0.0.46366

### Add-On Version

10.0.1 Beta

### Description

The custom color settings for the Shadowlands skin reset to the default when I installed the 10.0.1 Beta version of Masque.
While they can be changed back in the settings, the color choices don't actually get applied anymore.

-The default masque skins were not affected.
-The two other skins I have installed, Cainyx & Retina, were also not affected.

### Error Message

```text
No error message
```

### Steps to Reproduce

Install the new addon
Launch WoW

**Comment by StormFX** [comment]
I'll look into it.

**Comment by StormFX** [comment]
@nancikennedy I've been unable to reproduce this.

**Comment by nancikennedy** [comment]
I figured out why I was seeing this. The Bartender bars are only getting the color applied to buttons that have that slot filled. Empty slots are showing as the default color Shadowlands color. The bars I was looking at when I reported this were all empty at the time. I have Button Grid enabled on all the B4 bars, and am not hiding the borders on any of them

1st Row: Bartender Bars [1st: Partially filled, 2nd: Empty & 3rd: Partially filled, using Hex]
2nd Row: Farming Bar, empty.

The 3rd Bartender bar and the Farming Bar are acting as I would expect, but not the other two.

![Bartender](https://user-images.githubusercontent.com/8207666/199065629-855010cb-ab58-41dc-8b0a-35a03cc99597.JPG)

**Comment by StormFX** [comment]
The _Shadowlands_ skin has different texture and color set for empty buttons. If a button is empty, Masque checks to see if the skin has an empty texture. If it doesn't, it uses the `Normal` texture. The same applies to the color. If it's empty, it uses the skin's empty color if set, if not set, it uses the currently selected color for the `Normal` texture.

To solve this you can comment out the empty color (and if you want, the empty texture) in the skin file:

Line 64 from:
```lua
		EmptyColor = { 1, 1, 1, 1 }
```
To:
```lua
		-- EmptyColor = { 1, 1, 1, 1 }
```

Basically, just add two dashes before "E" in `EmptyColor`. The same with `EmptyTexture` if you want.

**Comment by nancikennedy** [comment]
Got it. Didn't realize that as I'd always had full bars before. Thanks for the help.
